storage/block/types.go

```diff
@@ -317,12 +315,24 @@ type Options interface {
 	// ClockOptions returns the clock options
 	ClockOptions() clock.Options
 
-	// SetDatabaseBlockAllocSize sets the databaseBlockAllocSize
+	// SetInstrumentOptions sets the instrument options
+	SetInstrumentOptions(value instrument.Options) Options
+
+	// InstrumentOptions returns the instrument options
+	InstrumentOptions() instrument.Options
+
+	// SetDatabaseBlockAllocSize sets the database block alloc size
 	SetDatabaseBlockAllocSize(value int) Options
 
-	// DatabaseBlockAllocSize returns the databaseBlockAllocSize
+	// DatabaseBlockAllocSize returns the database block alloc size
 	DatabaseBlockAllocSize() int
 
+	// SetWiredList sets the database block wired list
+	SetWiredList(value *WiredList) Options
+
+	// WiredList returns the database block wired list
+	WiredList() *WiredList
```

Comment:
General question: i don't know how uniform the load across the shards is, if we have hotspotting, is there value in making the LRUCache per shard?